### PR TITLE
Abstract classes refactor

### PR DESCRIPTION
**Pull request by aicioara**
Abstract classes are no longer abstract

### CODE REVIEW

File: src/components/abstractComponent.ts [hunk 1]
```diff
@@ -166,7 +165,7 @@ export module Components {
     public _doRender() {/* overwrite */}
 
     public _useLastCalculatedLayout(): boolean;
-    public _useLastCalculatedLayout(useLast: boolean) : AbstractComponent;
+    public _useLastCalculatedLayout(useLast: boolean) : Component;
```
Comment (bluong) [inline src/components/abstractComponent.ts]: Might as well take the chance to remove the space between ) and :

File: src/scales/quantitativeScale.ts [hunk 2]
```diff
@@ -1,15 +1,14 @@
 ///<reference path="../reference.ts" />
 
 module Plottable {
-export module Scales {
-  export class AbstractQuantitative<D> extends AbstractScale<D, number> {
+  export class Quantitative<D> extends Scale<D, number> {
```
Comment (jtlan) [inline src/scales/quantitativeScale.ts]: As @bluong mentioned, probably best to rename this to `QuantitativeScale`.

File: src/components/abstractComponent.ts (hunk 1 above)
Comment (aicioara) [inline src/components/abstractComponent.ts]: Not part of this CR. Opened #1912 for this

File: src/scales/quantitativeScale.ts (hunk 2 above)
Comment (aicioara) [inline src/scales/quantitativeScale.ts]: Done. Not pushing yet to not ruin the CR

**Comment by rcchen** [comment]
Nice!

**Pull request merged**